src/components/PDFView/index.tsx

```diff
@@ -0,0 +1,133 @@
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import variables from '@styles/variables';
+import * as CanvasSize from '@userActions/CanvasSize';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import PDFPasswordForm from './PDFPasswordForm';
+import type {PDFViewOnyxProps, PDFViewProps} from './types';
+
+function PDFView({onToggleKeyboard, fileName, onPress, isFocused, sourceURL, errorLabelStyles, maxCanvasArea, maxCanvasHeight, maxCanvasWidth, style}: PDFViewProps) {
+    const [isKeyboardOpen, setIsKeyboardOpen] = useState(false);
+    const styles = useThemeStyles();
+    const {windowHeight, isSmallScreenWidth} = useWindowDimensions();
+    const prevWindowHeight = useRef(windowHeight);
+    const {translate} = useLocalize();
+
+    /**
+     * On small screens notify parent that the keyboard has opened or closed.
+     *
+     * @param {Boolean} isKeyboardOpen True if keyboard is open
+     */
+    const toggleKeyboardOnSmallScreens = useCallback(
+        (isKBOpen: boolean) => {
+            if (!isSmallScreenWidth) {
+                return;
+            }
+            setIsKeyboardOpen(isKBOpen);
+            onToggleKeyboard?.(isKeyboardOpen);
+        },
+        [isKeyboardOpen, isSmallScreenWidth, onToggleKeyboard],
+    );
+
+    /**
+     * Verify that the canvas limits have been calculated already, if not calculate them and put them in Onyx
+     */
+    const retrieveCanvasLimits = () => {
+        if (!maxCanvasArea) {
+            CanvasSize.retrieveMaxCanvasArea();
+        }
+
+        if (!maxCanvasHeight) {
+            CanvasSize.retrieveMaxCanvasHeight();
+        }
+
+        if (!maxCanvasWidth) {
+            CanvasSize.retrieveMaxCanvasWidth();
+        }
+    };
+
+    useEffect(() => {
+        retrieveCanvasLimits();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    // shouldComponentUpdate(nextProps, nextState) {
+    //     return !_.isEqual(state, nextState) || !_.isEqual(props, nextProps);
+    // }
+
+    useEffect(() => {
+        if (!isKeyboardOpen && windowHeight < prevWindowHeight.current - 100) {
+            toggleKeyboardOnSmallScreens(true);
+        } else if (isKeyboardOpen && windowHeight > prevWindowHeight.current) {
+            toggleKeyboardOnSmallScreens(false);
+        }
+    }, [isKeyboardOpen, toggleKeyboardOnSmallScreens, windowHeight]);
+
+    const renderPDFView = () => {
+        const outerContainerStyle = [styles.w100, styles.h100, styles.justifyContentCenter, styles.alignItemsCenter];
+
+        return (
+            <View
+                style={outerContainerStyle}
+                tabIndex={0}
+            >
+                <PDFPreviewer
+                    contentContainerStyle={style}
```

Comment:
```suggestion
                    contentContainerStyle={style as CSSProperties}
```

Let's just assert here so we don't have to change the other places to use `CSSProperties`